pkg/utils/utils.go

```diff
@@ -70,3 +72,35 @@ func DecompressGZIP(compressedData []byte) (decompressedData []byte, err error)
 
 	return decompressedData, nil
 }
+
+// TODO: check if it's worth trying to compress the data prior to compressing it (e.g: using entropy algorithms)
+// This would save some CPU usage client side
+// TODO: have telemetry data type be passed in as a parameter to further check if we should compress data and which algorithm to use (e.g: deflate or gzip)
+func ShouldCompress(data []byte) (resultData []byte, compression *string, err error) {
+	// 'compression' is inserted as a sql.NullString, hence it is returned as a nullable string
+	var nilStr *string = nil
+	var validStr string = "gzip"
+
+	compressedData, err := CompressGZIP(data)
```

Comment:
Before attempting to compress the data it would probably be good idea to check the length of the input data, and skip compression if it is less than a certain size.

E.g.
```
% echo "" | gzip -fc | wc -c
21
echo "{}" | gzip -fc | wc -c
23
% echo '{"a":1}' | gzip -fc | wc -c
28
```
This tells us that the minimum data size worth trying to compress will be something > 23 characters, since the minimal valid JSON blob we can be given is an empty object `{}`, and the gzip compressed size of that will be 23 bytes.

With that in mind we probably don't want to try compressing anything less than say 64 bytes in length.

Probably define that min size as a constant before this routine, and do a check here and return the original data, and nilStr if the data length is less than that min size.